### DISCUSSION TITLE
Distributed training with pjit with convolutional layers

### DISCUSSION THREAD

**Discussion by aakashrkumar** [google/flax#2599] [Q&A] [answered]
Hi,
I was wondering what is the current ideal way to apply tensor parallelization to convolutional layers using pjit. What are the current best readable implementations to reference how to use pjit? 

Thanks

**Comment by cgarciae** [comment, marked as answer]
Hey @TheRealAakash, we've discussed this topic a bit internally, in the past we've pointed people to the [t5x](https://github.com/google-research/t5x) repo but sadly its a complex project. @IvyZX and @levskaya are working on a readable / simplified example which will serve as a reference. However, there are also talks of creating a `pjit` (Model Parallel) guide to Flax.

**Reply by aakashrkumar** [reply]
Thanks, I have looked at t5x, but one part that I am not sure about is how I would shard an image or convolutional layer
